schematools/permissions.py

```diff
@@ -105,7 +109,32 @@ def create_acl_from_profiles(engine, pg_schema, profile_list, role, scope):
                         engine.execute(grant_statement)
 
 
-def create_acl_from_schema(session, pg_schema, ams_schema, role, scope, dry_run, create_roles):
+def set_dataset_write_permissions(session, pg_schema, ams_schema, dry_run, create_roles):
+    grantee = f"write_{ams_schema.id}"
+    if create_roles:
+        _create_role_if_not_exists(session, grantee, dry_run=dry_run)
+    for table in ams_schema.get_tables(include_nested=True, include_through=True):
+        table_name = "{}_{}".format(
+            table.dataset.id, to_snake_case(table.id)
+        )  # een aantal table.id's zijn camelcase
+        table_privileges = ["INSERT", "UPDATE", "DELETE", "TRUNCATE", "REFERENCES"]
```

Comment:
De references is denk ik niet nodig. Omdat gebruikers met de write role alleen DML transacties zullen uitvoeren en geen verwijzingen zullen aanmaken

Reply:
Goed punt, hier twijfelde ik over. Zonder references staan we gebruikers niet toe om foreign key contraints toe te passen. We doen dat zelf ook niet, maar vroeg me af of er een reden is waarom we anderen dat ook niet willen toestaan.

Reply:
Geen `f"{table.dataset.id}_{to_snake_case(table.id)}"` ?